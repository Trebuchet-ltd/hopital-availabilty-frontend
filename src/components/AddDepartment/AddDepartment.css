.main{
    background-color: #f8f8f8f3!important;
    height: 100vh;
}
.sub {
    font-family: Nunito!important;
    background: #3e64ff!important;;
    color:#ffffff!important;
    text-transform: none!important;
    background: linear-gradient(180deg, #0338B9 0%, #3E64FF 100%)!important;
    border-radius: 10px!important;
}

.head-sec{
    background-color: #ffffff;
}
