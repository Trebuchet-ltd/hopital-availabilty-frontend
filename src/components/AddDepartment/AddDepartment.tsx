import {AuthComponent, AuthPropsLoc, AuthState} from "../../api/auth";
import {withRouter} from "react-router";
import {MenuItem, TextField, Button} from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";
import "./AddDepartment.css";
import {Department, DepartmentName, DepartmentNameObject} from "../../api/model";
import {toast} from "react-toastify";
import Skeleton from "@mui/material/Skeleton";

interface AddDepartmentState extends AuthState {
    departments: Array<DepartmentNameObject>,
    id: number,
    hospital: number,
    ready?: boolean
}


class AddDepartmentLoc extends AuthComponent<AuthPropsLoc, AddDepartmentState> 
{

    async componentDidMount() 
    {
        super.componentDidMount();
        
        const {hospital} = this.props.match.params as unknown as { hospital: number };
        const departments = await DepartmentName.filter();

        this.setState({
            departments: departments.results as Array<DepartmentNameObject>,
            hospital,
            ready: true,
        });
    }

    saveDepartment = () => 
    {
        Department.create({name_id: this.state.id, hospital: this.state.hospital})
            .then(() => 
            {
                this.props.history.push(`/details/${this.state.hospital}`);
                toast.success("thank you for the contribution", {
                    position: "bottom-center"
                });
            }).catch((error) => 
            {
                toast.error(error.details, {
                    position: "bottom-center"
                });
            });
    };

    render(): JSX.Element 
    {


        return (
            this.state.ready ?
                <div className="main h-100">
                    <div className="head-sec d-flex justify-content-between p-3 shadow-none h-25">
                        <CloseIcon onClick={() => this.props.history.goBack()} />
                        <p className="align-self-center m-0 p-0 justify-content-center"><b>Add Department details</b>
                        </p>
                        <Button className="sub" variant="contained" onClick={this.saveDepartment}>Submit</Button>
                    </div>

                    <div className="m-4">

                        <TextField className="mt-4" fullWidth variant="outlined" select label="Name"
                            InputLabelProps={{shrink: true, }} size="small"
                            onChange={({target}) => this.setState({id: Number(target.value)})}>
                            {this.state.departments.map(({name, id}, i) =>
                                <MenuItem value={id} key={i}>{name}</MenuItem>
                            )}
                        </TextField>

                    </div>

                </div> :
                <div className="main h-100">
                    <div className="head-sec d-flex justify-content-between p-3 shadow-none h-25">
                        <CloseIcon onClick={() => this.props.history.goBack()}/>
                        <p className="align-self-center m-0 p-0 justify-content-center"><b>Add Department details</b>
                        </p>
                        <Button className="sub" variant="contained" onClick={this.saveDepartment}>Submit</Button>
                    </div>

                    <div className="m-4">
                        <Skeleton variant="rectangular" className="mt-2 w-100"  height={118} />
                    </div>
                </div>

        );
    }
}

const AddDepartmentComponent = withRouter(AddDepartmentLoc);
export default AddDepartmentComponent
