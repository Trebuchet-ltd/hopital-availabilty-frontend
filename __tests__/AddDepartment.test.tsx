/**
 * @jest-environment jsdom
 */

import * as React from "react";
import { shallow } from "enzyme";
import {AddDepartmentComponent} from "../src/components/AddDepartment/AddDepartment";

describe("HandleAppState Component", () => {
    it("should render correctly by default", () => expect(shallow(<AddDepartmentComponent />)).toMatchSnapshot());
});

