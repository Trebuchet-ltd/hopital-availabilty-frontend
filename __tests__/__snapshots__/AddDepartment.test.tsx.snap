// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandleAppState Component should render correctly by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
